src/libasr/asr_verify.cpp

```diff
@@ -1090,6 +1090,15 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
         if( ASRUtils::get_asr_owner(x.m_derived_type) ) {
             symbol_owner = ASRUtils::symbol_name(ASRUtils::get_asr_owner(x.m_derived_type));
         }
+        
+        // Resolve the symbol in the current symbol table.
+        ASR::symbol_t* symbol = current_symtab->resolve_symbol(ASRUtils::symbol_name(x.m_derived_type));
+
+        // If the symbol is an ExternalSymbol, then omit the check.
+        if (ASR::is_a<ASR::ExternalSymbol_t>(*symbol)) {
+            return;
+        }
```

Comment:
I think this change is not correct --- I think the ASR is incorrect if the ExternalSymbol is not present in the current scope. So there is a bug in the way we construct the ASR that we need to fix.

Reply:
The symbol is there in the ``current_symtab`` imported as ``ExternalSymbol``. But, ``x.m_derived_type`` points to actual definition of the class which is inside another module, that is why ``symtab_in_scope`` fails. I think ASR is correct as the External Symbol is there in the current scope.

Reply:
The proper fix is described at https://github.com/lfortran/lfortran/issues/2957.